config/base.js

```diff
@@ -67,22 +67,21 @@ const SCHEMA_CONFIG_PRE_TEMPLATE_MERGE = Joi.object()
     .keys({
         template: Joi.string().regex(Regex.FULL_TEMPLATE_NAME_WITH_NAMESPACE),
         version: Joi.number().integer().min(1).max(50),
-        annotations: Annotations.annotations.when('template', { is: Joi.exist(), then: Joi.forbidden() }),
+        annotations: Annotations.annotations,
         jobs: SCHEMA_JOBS.when('template', { is: Joi.exist(), then: Joi.forbidden(), otherwise: Joi.required() }),
         shared: Joi.when('template', {
             is: Joi.exist(),
-            then: Joi.object().keys({
-                image: Job.image,
-                environment: Job.environment,
-                settings: Job.settings
+            then: SCHEMA_SHARED.append({
+                order: Joi.forbidden(),
+                steps: Joi.forbidden()
```

Comment:
Since we are not yet allowing customization of steps of the jobs inherited from the template and also not supporting defining new jobs, we should continue to selectively allow the fields that are supported. We should not allow customization of any field that changes the behavior of the jobs inherited from the template for now.

```suggestion
            then: Joi.object().keys({
                image: Job.image,
                environment: Job.environment,
                settings: Job.settings
```

Reply:
that was literally being asked from the issue `Support customization for other shared configs: annotations, blockedBy, cache, description, freezeWindows, parameters, provider, requires, secrets, sourcePaths, template`

Reply:
The task is to support customization for the specified fields in a broader sense. It does not imply that we should relax the schema to allow all these at once.

We need to analyze each field individually and define how the customization should be interpreted for various scenarios. Implement the logic to handle it in the backend before relaxing the schema.